transport/src/test/java/io/netty/channel/local/LocalChannelTest.java

```diff
@@ -794,6 +787,13 @@ public void initChannel(LocalChannel ch) throws Exception {
             final Promise<Void> assertPromise = cc.eventLoop().newPromise();
 
             cc.pipeline().addLast(new TestHandler() {
+                @Override
+                public Future<Void> connect(ChannelHandlerContext ctx,
+                                             SocketAddress remoteAddress, SocketAddress localAddress) {
+                    super.connect(ctx, remoteAddress, localAddress);
+                    return promise.setSuccess(null);
```

Comment:
This way we ignore the future from `connect` and end up calling `sync` on this future twice. Looks suspicious.